test/testdrive/materialized-view-refresh-options.td

```diff
@@ -383,4 +383,75 @@ true
   WHERE name = 'mv10';
 true
 
+# ----------------------------------------
+# Automated cluster scheduling for REFRESH
+# ----------------------------------------
+
+> CREATE CLUSTER scheduled_cluster (SIZE = '1', SCHEDULE = ON REFRESH, REPLICATION FACTOR = 1);
+
+# No MV yet, so the cluster should be turned off (even though it was turned on when we created it).
+> SELECT replication_factor FROM mz_catalog.mz_clusters WHERE name = 'scheduled_cluster';
+0
+
+> CREATE MATERIALIZED VIEW mv11
+  IN CLUSTER scheduled_cluster
+  WITH (REFRESH = EVERY '8 sec')
+  AS SELECT count(*) FROM t2;
+
+# The cluster should be turned on at some point.
+> SELECT replication_factor FROM mz_catalog.mz_clusters WHERE name = 'scheduled_cluster';
+1
+
+# And then the cluster should compute MV results.
+> SELECT * FROM mv11
+4
+
+# The cluster should be turned off at some point.
+> SELECT replication_factor FROM mz_catalog.mz_clusters WHERE name = 'scheduled_cluster';
+0
+
+> DELETE FROM t2;
+
+# The cluster should be turned on at some point again.
+> SELECT replication_factor FROM mz_catalog.mz_clusters WHERE name = 'scheduled_cluster';
+1
+
+# And then the cluster should compute MV results again.
+> SELECT * FROM mv11;
+0
+
+# Things should keep working if we switch from managed to unmanaged cluster and then back.
+> ALTER CLUSTER scheduled_cluster SET (MANAGED = false);
```

Comment:
What's the rationale here? It seems incorrect to allow users to create clusters that say `SCHEDULE = ON REFRESH, MANAGED = false` because they won't actually schedule on refresh.

Reply:
Unmanaged clusters will never have a `SCHEDULE` (and we already throw an error if the user tries to make one), and they will behave the same as `SCHEDULE = MANUAL`.

In this test, I just wanted to test what happens when we switch back from unmanaged to managed with `SCHEDULE = REFRESH`, but I forgot to make the test add back the `SCHEDULE = REFRESH` when switching back to managed.

More details on how things currently work:
- `ALTER CLUSTER scheduled_cluster SET (MANAGED = false);` simply makes us forget the `SCHEDULE`.
- `ALTER CLUSTER scheduled_cluster SET (MANAGED = false, SCHEDULE = ...);` throws an error.
- Trying to use `CREATE CLUSTER` to create an unmanaged cluster with `SCHEDULE = ON REFRESH` throws an error.

Reply:
I've fixed the test and added a test for the
`ALTER CLUSTER scheduled_cluster SET (MANAGED = false, SCHEDULE = ON REFRESH);`
error case.
b0300a90c0e7af2452d206bb535bbf23227795d4

Reply:
> ALTER CLUSTER scheduled_cluster SET (MANAGED = false); simply makes us forget the SCHEDULE.

Right, this seems like the bug to me! I don't think that command should silently change the schedule to `MANUAL`. If you don't specify `SCHEDULE` in the `ALTER` command, what you're indicating is that you want to alter the cluster's `MANAGED` property without changing any of its other properties—and if any of those other properties are incompatible with `MANAGED = false`, it should be an error. What we've done in other places (e.g., the "cc" migration with the `DISK` flag) is require users to explicitly type out both options:

```
ALTER CLUSTER scheduled_cluster SET (MANAGED = false, SCHEDULE + MANUAL);
```

Reply:
Ah, ok, that makes a lot of sense! I've fixed it and added a test for the new error case.